src/scripts/main.js

```diff
@@ -1 +1,92 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timerId = () => {
+    reject(new Error('First promise was rejected'));
+  };
+
+  document.addEventListener('click', () => {
+    resolve();
+    clearTimeout(timerId);
+  });
+
+  setTimeout(timerId, 3000);
+});
+
+const secondPromise = new Promise(resolve => {
+  document.addEventListener('click', () => {
+    resolve();
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve();
+  });
+});
+
+const thirdPromise = new Promise(resolve => {
+  let isClickedLeft = false;
+  let isClickedRight = false;
+
+  document.addEventListener('click', () => {
+    isClickedLeft = true;
+
+    // Проверяем условие при каждом клике
+    if (isClickedLeft && isClickedRight) {
+      resolve();
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    isClickedRight = true;
+
+    if (isClickedLeft && isClickedRight) {
+      resolve();
+    }
+  });
+});
+
+firstPromise
+  .then(() => {
+    const message = document.createElement('div');
+
+    message.setAttribute('data-qa', 'notification');
+    message.classList.add('success');
+    message.innerText = 'First promise was resolved';
+
+    document.body.append(message);
+
+    return secondPromise;
+  })
+  .catch(() => {
+    const message = document.createElement('div');
+
+    message.setAttribute('data-qa', 'notification');
+    message.classList.add('warning');
+    message.innerText = 'First promise was rejected';
+
+    document.body.append(message);
+
+    return secondPromise;
+  })
+  .then(() => {
+    const message = document.createElement('div');
+
+    message.setAttribute('data-qa', 'notification');
+    message.classList.add('success');
+    message.innerText = 'Second promise was resolved';
+
+    document.body.append(message);
+
+    return thirdPromise;
+  })
+  .then(() => {
+    const message = document.createElement('div');
+
+    message.setAttribute('data-qa', 'notification');
+    message.classList.add('success');
+    message.innerText = 'Third promise was resolved';
+
+    document.body.append(message);
+  });
```

Comment:
All promises should be independent
https://github.com/mate-academy/js_promises_practice_DOM/assets/61904995/79f42add-1637-449b-a5d8-ffa8fe1fe923